subsys/usb/device_next/class/usbd_cdc_ncm.c

```diff
@@ -0,0 +1,1372 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT zephyr_cdc_ncm_ethernet
+
+#include <zephyr/net/net_pkt.h>
+#include <zephyr/net/ethernet.h>
+
+#include <eth.h>
+
+#include <zephyr/usb/usbd.h>
+#include <zephyr/usb/usb_ch9.h>
+#include <zephyr/usb/class/usb_cdc.h>
+#include <zephyr/drivers/usb/udc.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(cdc_ncm, CONFIG_USBD_CDC_NCM_LOG_LEVEL);
+
+/* Set to 1 if you want to see hexdump of the packet in debug log level */
+#define DUMP_PKT 0
+
+#define CDC_NCM_ALIGNMENT          4U
+#define CDC_NCM_EP_MPS_INT         64U
+#define CDC_NCM_INTERVAL_DEFAULT   50000UL
+#define CDC_NCM_FS_INT_EP_INTERVAL USB_FS_INT_EP_INTERVAL(10000U)
+#define CDC_NCM_HS_INT_EP_INTERVAL USB_HS_INT_EP_INTERVAL(10000U)
+
+#define USB_SPEED_FS 12000000UL
+#define USB_SPEED_HS 480000000UL
```

Comment:
I think it is better to prefix this two macros with NCM_ or CDC_NCM_, otherwise it will likely there will be conflict soon or later.